### ISSUE TITLE
How should I adjust the hyperparameters to train a strong DDPM model based on CelebA for human face inpainting task?

### ISSUE THREAD

**Issue by LinWeiJeff** [jychoi118/ilvr_adm#33]
I'm very grateful for your work about this github repository!!! However, I have some questions about the hyperparameters to want to ask you.
I want to use [guided-diffusion](https://github.com/openai/guided-diffusion) to train a DDPM model based on CelebA dataset consisting of 120000 human face images (each image is 256(height)x256(width) pixels) for human face image synthesis task and then use the saved model to perform human face image inpainting task.

**Could you give me some suggestions about how to adjust the hyperparameters used in [guided-diffusion](https://github.com/openai/guided-diffusion) to train a DDPM model as strong as models provided by [guided-diffusion](https://github.com/openai/guided-diffusion) for human face image synthesis and then inpainting task?**
I wonder why the sizes of the models provided by [guided-diffusion](https://github.com/openai/guided-diffusion) are so big, especially why the size of [256x256 diffusion (not class conditional)](https://openaipublic.blob.core.windows.net/diffusion/jul-2021/256x256_diffusion_uncond.pt) is so big (about 2.1GB) and how it is trained.

I know the models provided by your repository are based on [P2-weighting](https://github.com/jychoi118/P2-weighting), and P2-weighting adds two extra hyperparameters (one is `p2_gamma` and the another is `p2_k`), but I still want to use [guided-diffusion](https://github.com/openai/guided-diffusion) to train my own DDPM model firstly because I want to learn DDPM from base and then advance, hence, I come here to ask you how to adjust the other hyperparameters used in [guided-diffusion](https://github.com/openai/guided-diffusion) except for p2_gamma and p2_k.

I hope the trained strong DDPM model learns the features of human face well so it can be used for human face image synthesis task and human face image inpainting task (i.e. recovering the masked parts of a masked human face image).

I want to know how to adjust the values of the hyperparameters in **diffusion_defaults() of script_util.py**, **model_and_diffusion_defaults() of script_util.py** and **create_argparser() of image_train.py** in [guided-diffusion](https://github.com/openai/guided-diffusion) to train a strong denoising model in DDPM based on my CelebA dataset.

I have tried some combinations of the hyperparameters used in [guided-diffusion](https://github.com/openai/guided-diffusion) to train, however, the human face image inpainting results of the saved model files `ema_0.9999_XXX.pt` and `modelXXX.pt` are both bad.
I mainly used [RePaint](https://github.com/andreas128/RePaint) to perform sampling for human face image inpainting task, as described in the README of RePaint, it uesd the pretrained model [celeba256_250000.pt](https://drive.google.com/uc?id=1norNWWGYP3EZ_o05DmoW1ryKuKMmhlCX) (The model is downloaded from [download.sh](https://github.com/andreas128/RePaint/blob/main/download.sh) and based on [guided-diffusion](https://github.com/openai/guided-diffusion)) to perform sampling for human face image inpainting task, and the size of the model is big (about 2.1GB) and its (celeba256_250000.pt) sampling results are not bad. However, I don't know why the model's size is so big and how it is trained.

In addition, because of the limitation of my GPU memory, I set the value of the hyperparameter `num_channels` only 64, I want to know if this hyperparameter affects the performance of the traind DDPM model. Should I try to set it larger?

In conclusion, I hope you can give me some suggestions about how to adjust the hyperparameters used in [guided-diffusion](https://github.com/openai/guided-diffusion) to let me get a strong DDPM model for human face image inpainting task.

Thanks a lot for your any help!!!
p.s. I directly and manually set up the values of the hyperparameters in the codes of [guided-diffusion](https://github.com/openai/guided-diffusion) not through any flag.
